sql/select.go

```diff
@@ -378,16 +384,45 @@ func (v *indexInfo) makeConstraints(exprs []parser.Exprs) {
 	startDone := false
 	endDone := false
 
-	for _, colID := range v.index.ColumnIDs {
-		constraint := indexConstraint{}
+	for i := 0; i < len(v.index.ColumnIDs); i++ {
+		colID := v.index.ColumnIDs[i]
+		var constraint indexConstraint
 
 		for _, e := range andExprs {
 			if c, ok := e.(*parser.ComparisonExpr); ok {
-				if q, ok := c.Left.(*qvalue); !ok || q.col.ID != colID {
-					// This expression refers to a column other than the one we're
-					// looking for.
-					continue
+				var tupleMap []int
+				switch t := c.Left.(type) {
+				case *qvalue:
+					if t.col.ID != colID {
+						// This expression refers to a column other than the one we're
+						// looking for.
+						continue
+					}
+
+				case parser.Tuple:
+					// If we have a tuple comparison we need to rearrange the comparison
+					// so that the order of the columns in the tuple matches the order in
+					// the index. For example, for an index on (a, b), the tuple
+					// comparison "(b, a) = (1, 2)" would be rewritten as "(a, b) = (2,
+					// 1)". Note that we don't actually need to rewrite the comparison,
+					// but simply provide a mapping from the order in the tuple to the
+					// order in the index.
+					idx := findColumnInTuple(t, colID)
```

Comment:
can you fold this into the loop below and do the `continue` in a separate `if len(tupleMap) == 0` after?